### PR TITLE
Plan window improvement

### PR DESCRIPTION
**Pull request by Cathyhjj**
Things to do before merging:

- [x] add tests
- [ ] write docs
- [ ] update iconfig_testing.toml
- [ ] flake8, black, and isort

### CODE REVIEW

**Comment by Cathyhjj** [comment]
do not review this PR yet, I wanted to add #329 in this PR too

**Comment by canismarko** [comment]
This numpy error is fixed in another PR: https://github.com/spc-group/haven/pull/339/commits/a8235951e3270553d05ed422785e853219fd59e7

The rest of the tests pass so I'll just review it as is.

File: src/firefly/plans/grid_scan.py [hunk 1]
```diff
@@ -108,14 +145,25 @@ def customize_ui(self):
...
-        # self.ui.detectors_list.selectionModel().selectionChanged.connect(
-        #     self.update_total_time
-        # )
         self.ui.spinBox_repeat_scan_num.valueChanged.connect(self.update_total_time)
         # Connect scan points change to update total time
         for region in self.regions:
             region.scan_pts_spin_box.valueChanged.connect(self.update_total_time)
+        self.ui.relative_scan_checkbox.stateChanged.connect(self.change_background)
+
+    def change_background(self, state):
```
Comment (canismarko) [inline src/firefly/plans/grid_scan.py:154]: Can this method be moved to the RegionsDisplay class so we don't need to write it twice? It looks identical between the line scan and grid scan versions.

File: src/firefly/plans/line_scan.py [hunk 2]
```diff
@@ -12,6 +13,7 @@
 
 
 class LineScanRegion(regions_display.RegionBase):
+    update_step_signal = Signal(int)
```
Comment (canismarko) [inline src/firefly/plans/line_scan.py:16]: What is this signal for? It looks like `LineScanDisplay.update_regions_step_size()` emits this signal, but all it does is connect directly to `LineScanRegion.update_step_size()`. Could `LineScanDisplay.update_regions_step_size()` call directly `LineScanRegion.update_step_size()`?

If we really need a signal, can it be renamed to describe what actually happened? Signals typically represent some changed state in the application (e.g. "checked", "textChanged" etc) instead of an action to take (that's a slot).

File: src/firefly/plans/line_scan.py [hunk 3]
```diff
@@ -54,6 +92,28 @@ def customize_ui(self):
             self.update_total_time
         )
         self.ui.spinBox_repeat_scan_num.valueChanged.connect(self.update_total_time)
+        self.ui.relative_scan_checkbox.stateChanged.connect(self.change_background)
+
+        # Connect scan_pts_spin_box value change to regions
+        self.ui.scan_pts_spin_box.valueChanged.connect(self.update_regions_step_size)
+
+    def update_regions_step_size(self, value):
```
Comment (canismarko) [inline src/firefly/plans/line_scan.py:100]: `"value"` is a bit vague. How about *step_size* or something like that?

File: src/firefly/tests/test_grid_scan_window.py [hunk 4]
```diff
@@ -59,7 +59,58 @@ async def test_time_calculator(display, sim_registry, ion_chamber):
...
+    # Check step size calculation for Region 1
+    assert region_1.step_size_line_edit.text() == "5.0"
+
+    # Test invalid input for Region 0
+    region_0.start_line_edit.setText("invalid")
+    region_0.update_step_size()
+    assert region_0.step_size_line_edit.text() == "N/A"
+
+    # Test edge case: num_points = 1 for Region 1
+    region_1.scan_pts_spin_box.setValue(1)
```
Comment (canismarko) [inline src/firefly/tests/test_grid_scan_window.py:96]: Good call testing for these edge cases!

**Review by canismarko — CHANGES_REQUESTED** [review]

File: src/firefly/plans/grid_scan.py (hunk 1 above)
Comment (Cathyhjj) [reply src/firefly/plans/grid_scan.py:154]: Good idea!

File: src/firefly/plans/line_scan.py (hunk 2 above)
Comment (Cathyhjj) [reply src/firefly/plans/line_scan.py]: Good idea, fixed, see the changed codes

File: src/firefly/plans/line_scan.py (hunk 3 above)
Comment (Cathyhjj) [reply src/firefly/plans/line_scan.py]: changed

**Review by canismarko — APPROVED** [review]
Looks good, feel free to merge when ready.
